types/execution/execution-triple.ts

```diff
@@ -0,0 +1,55 @@
+import {InstructionSet} from '../instructionsets.js';
+
+export enum ExecutionSpecialty {
+    cpu = 'cpu',
+    nvgpu = 'nvgpu',
+    amdgpu = 'amdgpu',
+}
+
+export class BaseExecutionTriple {
+    protected instructionSet: InstructionSet = 'amd64';
+    protected os: string = 'linux';
+    protected specialty: ExecutionSpecialty = ExecutionSpecialty.cpu;
+
+    getInstructionSet(): InstructionSet {
+        return this.instructionSet;
+    }
+
+    setInstructionSet(value: InstructionSet) {
+        this.instructionSet = value;
+    }
+
+    setOS(value: string) {
```

Comment:
I've been doing a bunch of JS and realise: you could use property setters/getters here for all these:

```ts
get os() { return this._os; }
set os(os: string) { this._os = os; }
```

then it "looks" like a normal field. More a flyby thought though than anything more.